tests/integration-tests/tests/performance_tests/test_startup_time.py

```diff
@@ -0,0 +1,126 @@
+        return False, distance
+    return True, distance
+
+
+def get_data(instance_type, os, cw_client):
+    data = []
+    dimensions = [{"Name": "InstanceType", "Value": instance_type}, {"Name": "OS", "Value": os}]
+
+    cluster_metrics = cw_client.list_metrics(
+        Namespace="ParallelCluster", MetricName="StartupTime", Dimensions=dimensions
+    )
+
+    logging.info(f"Metrics: {cluster_metrics}")
+
+    for metric in cluster_metrics["Metrics"]:
+        result = cw_client.get_metric_statistics(
+            Namespace="ParallelCluster",
+            MetricName="StartupTime",
+            Dimensions=metric["Dimensions"],
+            StartTime=datetime.now() - relativedelta(years=1),
+            EndTime=datetime.now(),
+            Period=30000,
+            Statistics=["Average"],
+            Unit="None",
+        )
+
+        logging.info(f"Results: {result}")
+
+        if result["Datapoints"]:
+            value = result["Datapoints"][0].get("Average")
+            data.append(value)
+    return data
+
+
+def get_metric(os, cluster, instance_type, instance_id, cw_client):
+    value = None
+
+    dimensions = [
+        {"Name": "OS", "Value": os},
+        {"Name": "InstanceID", "Value": instance_id},
+        {"Name": "ClusterName", "Value": cluster.name},
+        {"Name": "InstanceType", "Value": instance_type},
+    ]
+
+    result = cw_client.get_metric_statistics(
+        Namespace="ParallelCluster",
+        MetricName="StartupTime",
+        Dimensions=dimensions,
+        StartTime=datetime.now() - relativedelta(years=1),
+        EndTime=datetime.now(),
+        Period=30000,
+        Statistics=["Average"],
+        Unit="None",
+    )
+
+    logging.info(f"Results: {result}")
+
+    if result["Datapoints"]:
+        value = result["Datapoints"][0].get("Average")
+    return value
+
+
+def test_startup_time(pcluster_config_reader, clusters_factory, test_datadir, region, instance, os, scheduler):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+
+    cw_client = boto3.client("cloudwatch", region_name=region)
+
+    instances = describe_cluster_instances(
+        cluster.name,
+        region,
+        filter_by_node_type="Compute",
+    )
+
+    performance_degradation = {}
+
+    for instance in instances:
+        instance_type = instance["InstanceType"]
+        instance_id = instance["InstanceId"]
+        logging.info(f"Type: {instance_type}")
+
+        # get new startup time
+
+        value = get_metric(os, cluster, instance_type, instance_id, cw_client)
+        logging.info(f"Value: {value}")
+
+        # get historical data
+        data = get_data(instance_type, os, cw_client)
+        if value in data:
+            data.remove(value)
+
+        logging.info(f"Data of {instance_type}: {data}")
+
+        # evaluate data
+        if len(data) > MINIMUM_DATASET_SIZE and value:
```

Comment:
I `value` is None the test should fail because it was not able to collect the required data.
Furthermore, I think that the required sample size should be enforced on the number of compute nodes (eg: if we require a sample of size 10, then we should create a cluster with 10 compute nodes). With this enforcement is not even required to check if the length of data is greater than sample size, but we simply need to wait for the cluster to be in create complete.

Also, our tests should always yuse the same sample size, so we do not expect any case where data has length greater than the rewuired sample size.